langchain/src/llms/openai.ts

```diff
@@ -373,6 +376,24 @@ export class OpenAI extends BaseLLM implements OpenAIInput, AzureOpenAIInput {
             }
           );
 
+      const requestEndTime = Date.now();
```

Comment:
PromptLayer specific code should not be in the main OpenAI LLM code path - can you move it into the `PromptLayerOpenAI` class?